pkg/tfpfbridge/tests/nested_attr_test.go

```diff
@@ -32,5 +37,20 @@ func TestNestedType(t *testing.T) {
 
 	testresType := res.Schema(testresTypeName).Type().TerraformType(ctx)
 
-	t.Logf("%v", testresType)
+	obj := testresType.(tftypes.Object).AttributeTypes["services"].(tftypes.List).ElementType.(tftypes.Object)
+	assert.True(t, obj.AttributeTypes["protocol"].Is(tftypes.String))
+}
+
+func TestNestedOptionals(t *testing.T) {
+	t.Skip("Skipped as possible issue with terraform-plugin-framework")
+	ctx := context.TODO()
+	nestedOptional := tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+		"nestedOptional": {
+			Optional: true,
+			Type:     types.StringType,
+		},
+	})
+	assert.True(t, nestedOptional.GetAttributes()["nestedOptional"].IsOptional())
+	// Projecting to tftypes.Object seems to forget the optional-ity of nestedOptional.
+	assert.Contains(t, nestedOptional.Type().TerraformType(ctx).(tftypes.Object).OptionalAttributes, "nestedOptional")
```

Comment:
Here's the issue that surprised me, TerraformType() projection lost the optionality of the attributes.